packages/contracts/deploy/30_upgrade_repo/31_upgrade_repo.ts

```diff
@@ -15,6 +18,14 @@ type SemVer = [number, number, number];
 const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   const [deployer] = await hre.ethers.getSigners();
   const productionNetworkName: string = getProductionNetworkName(hre);
+  const network = getNetworkNameByAlias(productionNetworkName);
+  if (network === null) {
+    throw new UnsupportedNetworkError(productionNetworkName);
+  }
+  const networkDeployments = getLatestNetworkDeployment(network);
+  if (networkDeployments === null) {
+    throw `Deployments are not available on network ${network}.`;
+  }
```

Comment:
This makes code way longer instead of using the non-null assertion operator `!`. 
Is it worth it @jordaniza?

Reply:
IMO the `!` operator should be used sparingly - throwing gives a clear error message rather than potentially passing a falsy value somewhere else in the programme. So yes I think it's better.